metaflow/plugins/argo/argo_live_run.py

```diff
@@ -0,0 +1,248 @@
+from metaflow.plugins.argo.argo_client import ArgoClient
+from metaflow.metaflow_config import KUBERNETES_NAMESPACE
+import time
+
+
+class ArgoLiveRun:  # TODO: make child class of LiveRun
+    """
+    This class takes in information to identify and trigger a run of a
+    Metaflow flow using the Argo plugin. The run is initialized without
+    triggering a run, but a run should be triggered immediately after
+    initialization. This class should be initialized only with the trigger_run
+    function. The flow to trigger is identified based on the parameter for the
+    Metaflow flow name. Users may use additional parameters to alter the run,
+    and wait for the run to finish.
+
+    The object allows users to access information relating to the triggered
+    run, including booleans for whether the run has been triggered, is running,
+    and was successful, as well as for failed steps, and exceptions.
+    """
+
+    def __init__(
+        self,
+        flow_name: str = None,
+        alt_flow_id_info: dict = None,
+        parameters: dict = None,
+        wait: bool = True,
+        wait_timeout: int = 30,  # in minutes TODO: determine proper timeout
```

Comment:
Can those `parameters`, `wait` and `wait_timeout` be moved to `trigger` function instead? Looks like no other functions are using them, and they are not needed as instance variable.

Reply:
One issue with making this change is that makes it possible for a user to trigger additional runs using the same LiveRun object. I don't think this should be possible.

The current design already allows for users to trigger multiple runs using the same object w/ the same parameters. Even this isn't ideal, but it's a result of how we are structuring it.

I think that instead having the trigger happen in __init__ would solve this, though that design is disfavored.

Is it ok if I don't make your suggested change?

Reply:
To clarify/distill my previous message, I think the issue here is that the trigger method should only be called a single time, so there is no benefit to waiting until trigger is called to input parameters. Also, trigger will be called immediately after the object is created, so it doesn't help the user to have more time to decide the params/wait.

Reply:
That said, I made the changes. It's not a big deal, and the upcoming re-design will give time to really think through the best structure.

Reply:
Agree with your concerns over the design here. This is turning LiveRun into a more generic class, and failing to enforce only 1 run per class. Let's sync on that topic.

I think it's ok to move wait and wait_timeout out of `__init__`. It's also possible to have a separate `wait` function defined, giving user flexibility to trigger and then wait whenever they are ready. It may be handy when user wants to trigger multiple flows and then wait on each of them.